src/text/parsers/decimal.rs

```diff
@@ -91,13 +88,21 @@ fn decimal_from_text_components(
     // we've just removed the underscores above, the `from_str` methods below should always
     // succeed.
     let magnitude: Magnitude = if magnitude_text.len() < MAX_U64_DIGITS {
-        u64::from_str(&magnitude_text)
-            .expect("parsing coefficient magnitude as u64 failed")
-            .into()
+        let value = u64::from_str(&magnitude_text)
+            .or_fatal_parse_error(
+                input,
+                "parsing coefficient magnitude as u64 failed",
+            )?
+            .1;
```

Comment:
🗺️ `.or_fatal_parse_error` returns an `IonParseError`. We've handled the error case with a `?` on line 95. The success case is an `(remaining_input: &str, value)` tuple. If `from_str` succeeded, there's no more input so the `remaining_input` is `""`. We can discard that and take the `value` using the tuple field access syntax: `.1`.